ios/Classes/SwiftAmplitudeFlutterPlugin.swift

```diff
@@ -1,252 +1,400 @@
 import Flutter
 import UIKit
-import Amplitude
+import AmplitudeSwift
 
+// swiftlint:disable type_body_length
 @objc public class SwiftAmplitudeFlutterPlugin: NSObject, FlutterPlugin {
+// swiftlint:enable type_body_length
+    var amplitude: Amplitude?
+    static let methodChannelName = "amplitude_flutter"
+
     public static func register(with registrar: FlutterPluginRegistrar) {
-        let channel = FlutterMethodChannel(name: "amplitude_flutter", binaryMessenger: registrar.messenger())
+        let channel = FlutterMethodChannel(name: methodChannelName, binaryMessenger: registrar.messenger())
         let instance = SwiftAmplitudeFlutterPlugin()
         registrar.addMethodCallDelegate(instance, channel: channel)
     }
 
-    public func getPropertiesFromArguments(_ callArguments: Any?) throws -> [String:Any]? {
-        if let arguments = callArguments, let data = (arguments as! String).data(using: .utf8) {
+    // swiftlint:disable cyclomatic_complexity
+    // swiftlint:disable function_body_length
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch call.method {
+        case "init":
+            guard let args = call.arguments as? [String: Any] else {
+                print("\(call.method) called but call.arguments type casting failed.")
+                return
+            }
+
+            do {
+                amplitude = Amplitude(configuration: try getConfiguration(args: args))
+            } catch {
+                print("Initialization failed.")
+            }
+
+            // Set library
+            amplitude?.add(plugin: FlutterLibraryPlugin())
```

Comment:
nit: instead of maintaining the SDK version in 3 places (flutter, android, swift). Maybe we could pass in the SDK version from Flutter in the call args here.

I release semantic release makes it easy to update the version in multiple locations, but still feel a single source of truth would be nice.

Reply:
I tried it at https://github.com/amplitude/Amplitude-Flutter/pull/174 and later found that default events will still have native side library so I reverted the change and started using plugin to work this around at https://github.com/amplitude/Amplitude-Flutter/pull/175.

Reply:
I think this would allow the native library to get the library/version from Flutter
```suggestion
            amplitude?.add(plugin: FlutterLibraryPlugin(library = args.library))
            // or
            // amplitude?.add(plugin: FlutterLibraryPlugin(sdkVersion = args.sdkVersion))
```

Reply:
Good idea! Updated at https://github.com/amplitude/Amplitude-Flutter/pull/176/commits/f7029acb36eb24e5dc357dc07fc2eeb786077cc2 
- Pass `library` from Flutter via method channel at `init` call
- Pass to library plugins on native sides
- Set library by library plugins